### PR TITLE
seed with vw model

### PR DESCRIPTION
**Pull request by apoorvreddy**
Fix for issue #1393 

Changes

1. Too many files open exception
2. Adds VW_SeedWithModel to the API

### CODE REVIEW

**Comment by JohnLangford** [comment]
@eisber can you take a look at this?

File: vowpalwabbit/parse_args.cc [hunk 1]
```diff
@@ -1321,6 +1321,8 @@ void parse_sources(vw& all, io_buf& model, bool skipModelLoad)
 {
   if (!skipModelLoad)
     load_input_model(all, model);
+  else
+    model.close_file();
```
Comment (eisber) [inline vowpalwabbit/parse_args.cc]: I'm surprised this didn't have a negative affect right now. Does this solve your exception?

File: vowpalwabbit/vwdll.cpp [hunk 2]
```diff
@@ -54,6 +54,14 @@ VW_DLL_MEMBER VW_HANDLE VW_CALLING_CONV VW_InitializeA(const char * pstrArgs)
   return static_cast<VW_HANDLE>(all);
 }
 
+VW_DLL_MEMBER VW_HANDLE VW_CALLING_CONV VW_SeedWithModel(VW_HANDLE handle, const char * extraArgs)
+{
+  string s(extraArgs);
+  vw* origmodel = static_cast<vw*>(handle);
```
Comment (eisber) [inline vowpalwabbit/vwdll.cpp]: did you check how the C++ CLI layer (e.g. C# binding) calls this. I vaguely remember that there is some free() required.

File: vowpalwabbit/vwdll.h [hunk 3]
```diff
@@ -67,6 +67,7 @@ VW_DLL_MEMBER VW_HANDLE VW_CALLING_CONV VW_Initialize(const char16_t * pstrArgs)
 #endif
 VW_DLL_MEMBER VW_HANDLE VW_CALLING_CONV VW_InitializeA(const char * pstrArgs);
 VW_DLL_MEMBER VW_HANDLE VW_CALLING_CONV VW_InitializeWithModel(const char * pstrArgs, const char * modelData, size_t modelDataSize);
+VW_DLL_MEMBER VW_HANDLE VW_CALLING_CONV VW_SeedWithModel(VW_HANDLE handle, const char * extraArgs);
```
Comment (eisber) [inline vowpalwabbit/vwdll.h]: @JohnLangford who is using this API?

Comment (JohnLangford) [reply vowpalwabbit/vwdll.h]: I assume @apoorvreddy ?  It's not yet in the master branch.

Comment (eisber) [reply vowpalwabbit/vwdll.h]: I was asking about the C API in general, not just this function

Comment (JohnLangford) [reply vowpalwabbit/vwdll.h]: I don't believe that is answerable.

**Comment by JohnLangford** [comment]
Merged this.  Thanks.

File: vowpalwabbit/parse_args.cc (hunk 1 above)
Comment (apoorvreddy) [reply vowpalwabbit/parse_args.cc]: Hi @eisber . Apologies for the late reply. Yes, this solved the exception :) .

File: vowpalwabbit/vwdll.h (hunk 3 above)
Comment (apoorvreddy) [reply vowpalwabbit/vwdll.h]: @eisber My company is using VW in production systems. And we're using this API :)

**Pull request merged**